pkg/sts.go

```diff
@@ -0,0 +1,38 @@
+package pkg
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+// func TakeRole(sess *session.Session, roleARN, sessionName string) (*sts.AssumeRoleOutput, error) {
+func TakeRole(sess *session.Session, roleARN, sessionName string, logger log.Logger) {
+	stsClient := sts.New(sess)
```

Comment:
This should go through the awsclient mock infra as well. That way you can add tests assuring role asumption